src/test/java/seedu/address/model/person/PersonTest.java

```diff
@@ -31,27 +31,14 @@ public void isSamePerson() {
         // null -> returns false
         assertFalse(ALICE.isSamePerson(null));
 
-        // different phone and email -> returns false
-        Person editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB).build();
-        assertFalse(ALICE.isSamePerson(editedAlice));
+        // same name, different attributes -> returns true
+        Person editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB)
+                .withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_HUSBAND).build();
+        assertTrue(ALICE.isSamePerson(editedAlice));
 
         // different name -> returns false
         editedAlice = new PersonBuilder(ALICE).withName(VALID_NAME_BOB).build();
         assertFalse(ALICE.isSamePerson(editedAlice));
```

Comment:
We can also consider these cases:
* name differs in case only
* name has leading/trailing spaces

Without the above, a future developer can add a `trim()` or a `toLowerCase()` to the logic without any changes to the test cases.

Reply:
> Without the above, a future developer can add a trim() or a toLowerCase() to the logic without any changes to the test cases.

Do you mean if the developer directly modifies the `isSamePerson()` method with something like `toLowerCase()`? Yes, i think it's good to add a *name differs in case only test* to ensure case-sensitivity. But if I'm not wrong the leading and trailing spaces would already be handled by `ArgumentTokenizer.java` when it parses the user input.

`ArgumentTokenizer.java` documentation reference:
```
2. Leading and trailing whitespaces of an argument value will be discarded.<br>
```

Reply:
> leading and trailing spaces would already be handled by ArgumentTokenizer.java

You are right, but `PersonTest` is a unit test; `isSamePerson()` should be tested independently of other components of the application, such as the `ArgumentTokenizer`. Therefore, `isSamePerson()` should not assume that there aren't leading and trailing spaces.

Reply:
Good points. Instead, should be add asserts inside the Person class to verify that fields don't contain any leading/trailing spaces? It can be a separate PR though.

Reply:
Ahh I see @Zhiyuan-Amos, I have added the two test cases for `isSamePerson()`. I am not sure if it's correct, but I have decided to manipulate/append the Strings in the test cases rather than add another variable to `CommandTestUtil.java` for same name with different case/trailing spaces.
For the leading/trailing spaces test, I realised that I cannot test for leading spaces since the first character of a `Name` cannot start with a whitespace.